python/packages/ai/teams/ai/planners/planner.py

```diff
@@ -6,14 +6,17 @@
 from __future__ import annotations
 
 from abc import ABC, abstractmethod
+from typing import Generic, TypeVar
 
 from botbuilder.core import TurnContext
 
 from ...state import TurnState
 from .plan import Plan
 
+StateT = TypeVar("StateT", bound=TurnState)
 
-class Planner(ABC):
+
+class Planner(Generic[StateT], ABC):
```

Comment:
since this is now a generic you will probably need to update the AI class and its options that contain instances of Planner

Reply:
good catch, thank you!